cmd/lib/runfile/cmd.go

```diff
@@ -279,6 +285,29 @@ func setUpBQLStmt(tb *bql.TopologyBuilder, bqlFile string) error {
 	return nil
 }
 
+func checkStates(tb *bql.TopologyBuilder, saveUDSList string) error {
+	if saveUDSList == "" {
+		return nil // save all UDS and pass to check
+	}
+	states, err := tb.Topology().Context().SharedStates.List()
+	if err != nil {
+		return err
+	}
+	notFoundFlag := false
+	for _, name := range strings.Split(saveUDSList, ",") {
+		if _, ok := states[name]; !ok {
+			err := fmt.Errorf("the UDS is not found")
+			tb.Topology().Context().ErrLog(err).WithField("uds", name).Error(
+				"Invalid UDS name")
+			notFoundFlag = true
+		}
+	}
+	if notFoundFlag {
+		return fmt.Errorf("the target UDS list includes invalid name")
```

Comment:
Instead of using a boolean flag, how about this?

```
var missing []string
... in the loop, append a name of a UDS to missing that doesn't exist in the topology ...
if len(missing) > 0 {
  return fmt.Errorf("the topology doesn't have UDSs: %v", strings.Join(missing, ","))
}
```

Then, I think the logging in `checkStates` isn't necessary now.

```
tb.Topology().Context().ErrLog(err).WithField("uds", name).Error("Invalid UDS name")
```